src/associations/has-many.js

```diff
@@ -330,32 +330,20 @@ class HasMany extends Association {
   async set(sourceInstance, targetInstances, options) {
     if (targetInstances === null) {
       targetInstances = [];
+      options = options ? { ...options, destroy: true } : { destroy: true };
```

Comment:
why set the `destroy` option here?

If you want to let the user decide whether to delete the disassociated entity, the option should be set by the user
If you want to decide automatically, `remove` should determine that itself by checking if the foreign key is nullable